python/ray/includes/unique_ids.pxi

```diff
@@ -212,15 +212,55 @@ cdef class ClientID(UniqueID):
         return <CClientID>self.data
 
 
-cdef class JobID(UniqueID):
+cdef class JobID(BaseID):
+    cdef CJobID data
 
     def __init__(self, id):
-        check_id(id)
+        check_id(id, CJobID.Size())
         self.data = CJobID.FromBinary(<c_string>id)
 
     cdef CJobID native(self):
         return <CJobID>self.data
 
+    @classmethod
+    def from_int(cls, value):
+        return cls(CJobID.FromInt(value).Binary())
+
+    @classmethod
+    def compute_job_id_from_driver(cls, driver_id):
+        assert isinstance(driver_id, WorkerID)
+        return cls(CJobID.ComputeJobIdFromDriver(
+            CWorkerID.FromBinary(driver_id.binary())).Binary())
+
+    @classmethod
+    def compute_driver_id_from_job(cls, job_id):
```

Comment:
Nit. I would consider making these 2 helper functions independent, instead of inside `JobID`. 
Because that's how we usually define util functions.